src/Type/MixedType.php

```diff
@@ -144,6 +144,16 @@ public function unsetOffset(Type $offsetType): Type
 		return $this;
 	}
 
+	public function getKeysArray(): Type
+	{
+		return AccessoryArrayListType::intersectWith(new ArrayType(new IntegerType(), new UnionType([new IntegerType(), new StringType()])));
+	}
+
+	public function getValuesArray(): Type
+	{
+		return AccessoryArrayListType::intersectWith(new ArrayType(new IntegerType(), new MixedType($this->isExplicitMixed)));
+	}
```

Comment:
I think these 2 could support a subtracted array type

Reply:
They are very forgiving and follow https://github.com/phpstan/phpstan-src/pull/1850#discussion_r996297966 now by returning an array already which is preciser than mixed ~array

Reply:
I think, taking `subtracted` into account it should get `*NEVER*` in this case
https://phpstan.org/r/918a37d5-92cf-4010-9dd8-d4ebbc315274

Reply:
Ah, I misunderstood you. That indeed makes sense

Reply:
isArray()->no() should be the check we need. Feel free to adapt it if you want, if not I can check it out in the evening or tomorrow 😊

Reply:
I will give it a try